exercises/matthew_bowles/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,108 @@
+#include "ShapeSorter.h"
+
+ShapeSorter::ShapeSorter()
+{
+}
+
+/** prints shapes that match a given type
+*
+* @param type The type of shape to find e.g. "Square"
+* @param shapes The vector of shape pointers to traverse
+*
+*/
+void ShapeSorter::matchType(const string& type, vector<Shape*> shapes)
+{
+	auto iter = shapes.begin();
+	while (iter != shapes.end())
+	{
+		if (type.compare((*iter)->getType()) != 0)
+			iter = shapes.erase(iter);
```

Comment:
You want to avoid erasing in a vector. See [here](https://stackoverflow.com/questions/4663455/why-is-erase-function-so-expensive).

A solution would be to populate  a new vector with pointers with the shapes of interest. Also try to avoid too many iterators. Range based for loops are more readable.

You can do simple string comparison with `==`